opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,172 @@
+        aggregation: Optional[Type[Aggregation]] = None,
+    ):
+        """
+        An instance of `View` is an object that can perform two actions:
+
+        1. Match instruments: When an instrument matches a view, measurements
+           received by that instrument will be processed.
+        2. Customize metric streams: A metric stream is identified by a match
+           between a view and an instrument and a set of attributes. The metric
+           stream can be customized by certain attributes of the corresponding
+           view.
+
+        The attributes documented next serve one of the previous two purposes.
+
+        Args:
+            instrument_type: This is an instrument matching attribute: the class
+            the instrument must be to match the view.
+            instrument_name: This is an instrument matching attribute: the name
+            the instrument must have to match the view.
+            meter_name: This is an instrument matching attribute: the name
+            the instrument meter must have to match the view.
+            meter_version : This is an instrument matching attribute: the
+            version the instrument meter must have to match the view.
+            meter_schema URL : This is an instrument matching attribute: the
+            schema URL the instrument meter must have to match the view.
+            name: This is a metric stream customizing attribute: the name of
+            the metric stream. If `None`, the name of the instrument will be
+            used.
+            description: This is a metric stream customizing attribute: the
+            description of the metric stream. If `None`, the description of the
+            instrument will be used.
+            attribute_keys: This is a metric stream customizing attribute: this
+            is a set of attribute keys. If not `None` then only the measurement
+            attributes that are in `attribute_keys` will be used to identify
+            the metric stream.
+            aggregation: This is a metric stream customizing attribute: every
+            metric stream has an aggregation instance, this is the class of
+            aggregation this instance will be. If `None` the default
+            aggregation class of the instrument will be used.
+        """
+
+        if (
+            instrument_type
+            is instrument_name
+            is meter_name
+            is meter_version
+            is meter_schema_url
+            is None
+        ):
+            raise Exception(
+                "Some instrument selection criteria must be provided"
+            )
+
+        self._instrument_type = instrument_type
+        self._instrument_name = instrument_name
+        self._meter_name = meter_name
+        self._meter_version = meter_version
+        self._meter_schema_url = meter_schema_url
+
+        self._name = name
+
+        if self._name is not None:
+            if self._instrument_type is not None:
```

Comment:
Curious as to why warnings are logged if some properties are missing?

Reply:
Per the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#view):

> the SDK MUST NOT allow Views with a specified name to be declared with instrument selectors that select more than one instrument (e.g. wild card instrument name).

Reply:
I see you've changed this to raise an `Exception` so that's fine. My next question is, how does the current logic check for the excerpt that you outlined from the spec? It looks like the current logic makes it so that if `self._name` is defined, no other filtering property can be defined? How does that imply the above spec statement? I might be confused as to what a defined `name` implies.

Reply:
> My next question is, how does the current logic check for the excerpt that you outlined from the spec? It looks like the current logic makes it so that if `self._name` is defined, no other filtering property can be defined? How does that imply the above spec statement?

Well, I think the current code matches the spec statement. :shrug:  Can you point out where specifically is a discrepancy?

Reply:
> In order to avoid conflicts, views which specify a name SHOULD have an instrument selector that selects at most one instrument.

@lzchen if we were to allow `v1 = View(name="name", instrument_type=Counter)` that would allow all counter instruments to be matched which would be conflicting, right? This is my current understanding.

Reply:
@lzchen @lonewolf3739 please take a look again, I have refactored the matching method :v: